pe.c

```diff
@@ -772,8 +905,22 @@ handle_image (void *data, unsigned int datasize,
 			clear_attrs &= ~MEM_ATTR_X;
 		}
 		update_mem_attrs(addr, length, set_attrs, clear_attrs);
-	}
 
+		if ((Section->Characteristics & EFI_IMAGE_SCN_CNT_CODE ||
+		     Section->Characteristics & EFI_IMAGE_SCN_CNT_INITIALIZED_DATA) &&
+		    Section->Characteristics & EFI_IMAGE_SCN_MEM_READ) {
+			efi_status = cache_section(li->ParentHandle, Section->Name, base, raw_length);
```

Comment:
This was actually my mistake, I marked the earlier comment as solved, it was right the first time with `li`